apps/server/src/modules/user-login-migration/service/start-user-login-migration-check.service.spec.ts

```diff
@@ -0,0 +1,206 @@
+				},
+			],
+		}).compile();
+
+		service = module.get(StartUserLoginMigrationCheckService);
+		userLoginMigrationService = module.get(UserLoginMigrationService);
+		authorizationService = module.get(AuthorizationService);
+		schoolService = module.get(SchoolService);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	describe('checkPreconditions', () => {
+		describe('when preconditions are met', () => {
+			const setup = () => {
+				const userId = 'userId';
+
+				const migration: UserLoginMigrationDO = new UserLoginMigrationDO({
+					schoolId: 'schoolId',
+					targetSystemId: 'targetSystemId',
+					startedAt: new Date(),
+				});
+
+				const school: SchoolDO = schoolDOFactory.buildWithId({ id: migration.id });
+
+				const context = {
+					action: Action.write,
+					requiredPermissions: [Permission.USER_LOGIN_MIGRATION_ADMIN],
+				};
+
+				authorizationService.checkPermission.mockReturnThis();
+				userLoginMigrationService.findMigrationBySchool.mockResolvedValue(null);
+				schoolService.getSchoolById.mockResolvedValue(school);
+
+				return { userId, migration, schoolId: school.id as string, school, context };
+			};
+
+			it('should call checkPermission', async () => {
+				const { userId, school, context } = setup();
+
+				await service.checkPreconditions(userId, school.id as string);
+
+				expect(authorizationService.checkPermission).toHaveBeenCalledWith(userId, school, context);
+			});
+
+			it('should get school by id ', async () => {
+				const { userId, schoolId } = setup();
+
+				await service.checkPreconditions(userId, schoolId);
+
+				expect(schoolService.getSchoolById).toHaveBeenCalledWith(schoolId);
+			});
+
+			it('should check if migration exists', async () => {
+				const { userId, schoolId } = setup();
+
+				await service.checkPreconditions(userId, schoolId);
+
+				expect(userLoginMigrationService.findMigrationBySchool).toHaveBeenCalledWith(schoolId);
+			});
+		});
+
+		describe('when school has no officialSchoolNumber', () => {
+			const setup = () => {
+				const school: SchoolDO = schoolDOFactory.buildWithId();
+				school.officialSchoolNumber = undefined;
+
+				const user: User = userFactory.buildWithId();
+
+				const migration: UserLoginMigrationDO = new UserLoginMigrationDO({
+					schoolId: 'schoolId',
+					targetSystemId: 'targetSystemId',
+					startedAt: new Date(),
+				});
+
+				schoolService.getSchoolById.mockResolvedValue(school);
+				authorizationService.getUserWithPermissions.mockResolvedValue(user);
+				authorizationService.checkPermission.mockReturnThis();
+				schoolService.getSchoolById.mockResolvedValue(school);
+
+				userLoginMigrationService.findMigrationBySchool.mockResolvedValue(null);
+				userLoginMigrationService.startMigration.mockResolvedValue(migration);
+
+				return { user, migration, schoolId: school.id as string };
+			};
+
+			it('should throw ForbiddenException ', async () => {
```

Comment:
your description does not fit you exception.
For more readability you can seperate you function and you expection.
const func = () = > service.checkPreconditions(user.id, schoolId)

and then 

await expect(func()) ....